cluster/src/main/scala/org/apache/spark/sql/streaming/SnappyStreamingContextFactory.scala

```diff
@@ -61,13 +56,18 @@ class SnappyStreamingContextFactory extends SparkContextFactory {
     new SnappyStreamingContext(LeadImpl.getInitializingSparkContext,
       Milliseconds(interval)) with ContextLike {
 
+      private val addedJars = scala.collection.mutable.ArrayBuffer.empty[String]
```

Comment:
New Spark streaming will add additional complexities. Its a good point you raised. Structured streaming does not require an explicit streaming context to be created. A simple SparkSession ( and hence SnappySession) is enough to start a streaming job.

Theoretically it can be as easy as 
````object SomeStreamJob extends SnappySQLJob {
override def runSnappyJob(snc: SnappySession, jobConfig: Config): Any = {
val lines = snc.readStream
      .format("socket")
      .option("host", host)
      .option("port", port)
      .load().as[String]
}`

Should we explicitly stop starting streaming jobs in this fashion or we should support it. Supporting this would means out job jar clean up code needs more work. We can not clean a job jar ( from executors)  if its a streaming job . Then the question remains how to handle same class multiple versions ( for multiple stream jobs) ? Also multiple stream sessions can run concurrently. 

Also another problem is how to stop a running query. It is a query.stop and not a global one like StreaminngContext.stop(). That means query reference needs to be stored in job server against some id and that id needs to be returned ( shown in UI) to users. At some later point of time one can come and stop a query. Need to see if this is being worked upon with job-server. If this can be achieved class related problem can be taken care of.